content/courses/program-security/type-cosplay.md

```diff
@@ -236,99 +239,97 @@ pub struct AdminInstruction<'info> {
 }
 
 #[account]
+#[derive(InitSpace)]
 pub struct AdminConfig {
     admin: Pubkey,
 }
 
 #[account]
+#[derive(InitSpace)]
 pub struct UserConfig {
     user: Pubkey,
 }
 ```
 
-It’s important to note that this is a vulnerability you don’t really have to
-worry about when using Anchor - that’s the whole point of it in the first place!
-After going through how this can be exploited if not handled properly in a
-native rust program, hopefully you have a much better understanding of what the
-purpose of the account discriminator is in an Anchor account. The fact that
-Anchor sets and checks this discriminator automatically means that developers
-can spend more time focusing on their product, but it’s still very important to
-understand what Anchor is doing behind the scenes to develop robust Solana
-programs.
+This vulnerability is something you generally don’t have to worry about when
+using Anchor—that's the whole point! However, after exploring how this issue can
+arise in native Rust programs, you should now have a better understanding of the
+importance of the account discriminator in an Anchor account. Anchor's automatic
+discriminator checks mean that developers can focus more on their product, but
+it's still crucial to understand what Anchor is doing behind the scenes to build
+robust Solana programs.
 
 ## Lab
 
-For this lab we’ll create two programs to demonstrate a type cosplay
-vulnerability.
+In this lab, you'll create two programs to demonstrate a type cosplay
+vulnerability:
 
-- The first program will initialize program accounts without a discriminator
-- The second program will initialize program accounts using Anchor’s `init`
-  constraint which automatically sets an account discriminator
+- The first program initializes accounts without a discriminator.
+- The second program initializes accounts using Anchor's `init` constraint,
+  which automatically sets an account discriminator.
 
-#### 1. Starter
+### 1. Starter
 
-To get started, download the starter code from the `starter` branch of
-[this repository](https://github.com/Unboxed-Software/solana-type-cosplay/tree/starter). The
-starter code includes a program with three instructions and some tests.
+To get started, download the starter code from the starter branch of
+[this repository](https://github.com/Unboxed-Software/solana-type-cosplay/tree/starter).
```

Comment:
Please use https://github.com/solana-developers/type-cosplay for the github link (it has your changes).

Reply:
Sure will use this.

Reply:
Updated link for both starter and solution branches.